src/backend/tools/google_search.py

```diff
@@ -0,0 +1,70 @@
+from typing import Any, Dict, List
+
+from googleapiclient.discovery import build
+
+from backend.config.settings import Settings
+from backend.crud import agent_tool_metadata as agent_tool_metadata_crud
+from backend.database_models.database import DBSessionDep
+from backend.schemas.context import Context
+from backend.tools.base import BaseTool
+
+
+class GoogleWebSearch(BaseTool):
+    NAME = "google_web_search"
+    API_KEY = Settings().tools.google_web_search.api_key
+    CSE_ID = Settings().tools.google_web_search.cse_id
+
+    def __init__(self):
+        self.client = build("customsearch", "v1", developerKey=self.API_KEY)
+
+    @classmethod
+    def is_available(cls) -> bool:
+        return cls.API_KEY is not None and cls.CSE_ID is not None
+
+    def get_filtered_domains(self, session: DBSessionDep, ctx: Context) -> list[str]:
```

Comment:
this seems duplicated from tavily.py. you might want to extract this into a mixin:
```python
class DomainFilter:
  def get_filtered_domains(...)
  
class GoogleWebSearch(BaseTool, DomainFilter):
   pass

class TavilyInternetSearch(BaseTool, DomainFilter):
  pass
```

Or just extract it into a plain function somwhere and call it from both files.

Reply:
@raphael-cohere yup I'm planning to do this in a later PR, since Brave/Google will implement this. I can handle this